rueidisotel/trace.go

```diff
@@ -61,6 +61,32 @@ func WithDBStatement(f StatementFunc) Option {
 // StatementFunc is a the function that maps a command's tokens to a string to put in the db.statement attribute
 type StatementFunc func(cmdTokens []string) string
 
+type commandMetrics struct {
+	duration   metric.Float64Histogram
+	errors     metric.Int64Counter
+	addOpts    []metric.AddOption
+	recordOpts []metric.RecordOption
+	opAttr     bool
+}
+
+func (c commandMetrics) recordDuration(ctx context.Context, op string, now time.Time) {
+	opts := c.recordOpts
+	if c.opAttr {
+		opts = append(c.recordOpts, metric.WithAttributes(attribute.String("operation", op)))
+	}
+	c.duration.Record(ctx, time.Since(now).Seconds(), opts...)
+}
+
+func (c commandMetrics) recordError(ctx context.Context, op string, err error) {
```

Comment:
```suggestion
func (c *commandMetrics) recordError(ctx context.Context, op string, err error) {
```

Reply:
We should use a pointer receiver to avoid unnecessary copying of the structure here.

Reply:
Yep, makes sense.